core/app/finders/spree/products/find.rb

```diff
@@ -324,6 +339,20 @@ def order_by_price(scope, sort_order)
       def order_by_best_selling(scope)
         scope.by_best_selling(:desc)
       end
+
+      def products_matching_all_taxonomies_ids(taxonomies_products, taxonomies)
+        all_product_ids = taxonomies_products.map(&:id)
+        taxon_groups = taxonomies.values.map { |v| v['taxon_ids'].map(&:to_i) }
```

Comment:
I don't think we can map `taxon_ids` to `integer`, Spree can have string IDs as well